app/services/serialize_offer.rb

```diff
@@ -8,10 +8,14 @@ def initialize(application_choice:, standard_conditions:, further_condition_attr
   end
 
   def save
-    @offer = Offer.find_or_create_by(application_choice: application_choice)
+    ActiveRecord::Base.transaction do
+      @offer = Offer.find_or_create_by(application_choice: application_choice)
 
-    serialize_standard_conditions
-    serialize_further_conditions
+      serialize_standard_conditions
+      serialize_further_conditions
+    end
```

Comment:
This way the service can be used on its own. In the case that it's used through Make or ChangeOffer, a nested transaction block becomes part of the parents transaction block